publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md

```diff
@@ -0,0 +1,167 @@
+
+<!--
+//
+//	Copyright 2019, 2020 IBM Corporation and others.
+//
+//	Licensed under the Apache License, Version 2.0 (the "License");
+//	you may not use this file except in compliance with the License.
+//	You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+//	Unless required by applicable law or agreed to in writing, software
+//	distributed under the License is distributed on an "AS IS" BASIS,
+//	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//	See the License for the specific language governing permissions and
+//	limitations under the License.
+//
+-->
+
+## Introduction
+
+This guide will walk you through the steps needed to run a pipeline to build your application, publish the image to a registry, and optionally deploy the application on your cluster.
+
+For this guide, we will use the one of the default pipelines, `java-openliberty-build-deploy-pl`, that is activated by the Kabanero operator. You can replace this with any of the other `build-deploy` pipelines or your custom pipeline.
+
+## Steps
+
+1. Clone the `https://github.com/kabanero-io/kabanero-pipelines/` repository.
+   * cd into the `pipelines/sample-helper-files` directory
+2. Make sure you have the appropriate storage necessary to drive pipelines. If you have a dynamic storage provisioner available on your cluster, you can proceed to the next step. If not, you can set up a simple NFS-based persistent volumen (PV) using the nfs-pv.yaml sample provided.
+   * Note: you must update the IP address before applying the file.
+   * Run `oc apply nfs-pv.yaml`
+3. If you are using a private GitHub repo, create the secret for your repo. Follow the steps in Secrets section of [Working with Pipelines](../working-with-pipelines/working-with-pipelines.html#getting-started).
+4. Determine whether it is necessary to create the secret for your image registry.
+   * If you are using the internal route of the internal registry of your OCP cluster, you do not have to configure a secret as long as the service account you are running the pipeline with has appropriate permissions, such as the `kabanero-pipeline` service account.
+   * If you are using the external route of the internal registry of your OCP cluster, you do not have to configure a secret. The pipeline will recognize the URL and translate it to the internal route of the registry.
+   * If you are using a private registry, follow the steps in Secrets section of [Working with Pipelines](../working-with-pipelines/working-with-pipelines.html#getting-started) to set up a secret for your registry.
+5. Set up certificates or specify insecure connection for the registry access. For instructions and additional information, see Transport layer security (TLS) verification for image registry access in pipelines in [Working with pipelines](../working-with-pipelines/working-with-pipelines.html#transport-layer-security-tls-verification-for-image-registry-access-in-pipelines).
```

Comment:
- Set up certificates or specify an insecure connection...? 
- see **Transport layer security (TLS) verification for image registry access in pipelines** in .....(makes it easier to read)